src/api/app/controllers/webui/webui_controller.rb

```diff
@@ -250,33 +250,36 @@ def lockout_spiders
 
   def check_user
```

Comment:
An empty HTTP_X_USERNAME (how can this happen?) is easily checked with .blank in line 259. No need to rewrite the whole logic IMHO...